subprojects/core-api/src/main/java/org/gradle/api/provider/ListProperty.java

```diff
@@ -30,4 +32,27 @@
  */
 @Incubating
 public interface ListProperty<T> extends PropertyState<List<T>> {
+    /**
+     * Adds an element to the property.
```

Comment:
We may want to call this out as being 'lazy' (in that it doesn't cause the existing provider to be evaluated) and is provided as a convenience over calling `add(Provider<T>)` yourself.

Reply:
I will make the comment clearer.